### PR TITLE
[SPARK-25479][TEST] Refactor DatasetBenchmark to use main method

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Refactor `DatasetBenchmark` to use main method.
Generate benchmark result:
```sh
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.DatasetBenchmark"
```

## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
@dongjoon-hyun I think this refactor is ready to go. Thanks.

File: sql/core/src/test/scala/org/apache/spark/sql/DatasetBenchmark.scala [hunk 1]
```diff
@@ -242,75 +248,20 @@ object DatasetBenchmark {
...
 
-  def main(args: Array[String]): Unit = {
-    val spark = SparkSession.builder
-      .master("local[*]")
-      .appName("Dataset benchmark")
-      .getOrCreate()
+  val spark = SparkSession.builder
+    .master("local[*]")
+    .appName("Dataset benchmark")
+    .getOrCreate()
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/DatasetBenchmark.scala]: Can we move this SparkSession building part into `benchmark()` function and before `runBenchmark("Dataset Benchmark")`?

File: sql/core/src/test/scala/org/apache/spark/sql/DatasetBenchmark.scala [hunk 2]
```diff
@@ -242,75 +249,22 @@ object DatasetBenchmark {
     benchmark
   }
 
-  def main(args: Array[String]): Unit = {
-    val spark = SparkSession.builder
+  override def getSparkSession: SparkSession = {
```
Comment (wangyum) [inline sql/core/src/test/scala/org/apache/spark/sql/DatasetBenchmark.scala]: Need override default `SparkSession` as default `SparkSession` is:
```scala
    SparkSession.builder()
      .master("local[1]")
      .appName(this.getClass.getCanonicalName)
      .config(SQLConf.SHUFFLE_PARTITIONS.key, 1)
      .config(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key, 1)
      .getOrCreate()
```

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
@wangyum . Could you review and merge https://github.com/wangyum/spark/pull/13 ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (pending Jenkins)

**Comment by dongjoon-hyun** [comment]
Hi, @jiangxb1987 .
Could you review (and merge) this PR?

**Comment by wangyum** [comment]
Congratulation, @jiangxb1987

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**